### PR TITLE
Add window size controls

### PR DESCRIPTION
**Pull request by aomarks**
Fixes https://github.com/Polymer/tachometer/issues/69

Previously we had no control over window size, and we got whatever the browser wanted by default. This PR adds the `--window-size` flag and `browser: {windowSize: {width, height} }` JSON config file formats to control the size. Defaults to 1024x768.

Also required a little refactoring so that we can represent browser options in a more structured way, both internally and in the JSON config file.

Manually tested with all browsers on Windows.

### CODE REVIEW

File: src/browser.ts [hunk 1]
```diff
@@ -90,11 +120,12 @@ export function parseAndValidateBrowser(str: string): BrowserConfig {
   if (headless === true && !headlessBrowsers.has(name)) {
     throw new Error(`Browser ${name} does not support headless mode.`);
   }
-  const config: BrowserConfig = {name, headless};
-  if (remoteUrl !== undefined) {
-    config.remoteUrl = remoteUrl;
+  if (remoteUrl !== undefined && !isHttpUrl(remoteUrl)) {
+    throw new Error(`Invalid browser remote URL "${remoteUrl}".`);
+  }
+  if (windowSize.width < 0 || windowSize.height < 0) {
```
Comment (rictic) [inline src/browser.ts]: This checks < but the message says <=

The JSON schema says minimum of 0, so < would be the most consistent.

(I'm standing up for the zero-sized browser community here. It's the next step beyond no-js, no-images, no-css. just embed a WAV file in an `<audio>` tag)

File: src/browser.ts [hunk 2]
```diff
@@ -134,6 +174,11 @@ function firefoxOpts(config: BrowserConfig): firefox.Options {
     // tslint:disable-next-line:no-any TODO Incorrect types.
     (opts as any).addArguments('-headless');
   }
+  const {width, height} = config.windowSize;
+  // tslint:disable-next-line:no-any TODO Incorrect types.
+  (opts as any).addArguments(`-width=${width}`);
```
Comment (rictic) [inline src/browser.ts]: one dash? what is this anarchy

File: src/browser.ts [hunk 3]
```diff
@@ -158,6 +203,11 @@ export async function openAndSwitchToNewTab(driver: webdriver.WebDriver):
     throw new Error(`Expected to create 1 new tab, got ${newTabs.length}`);
   }
   await driver.switchTo().window(newTabs[0]);
+  if (config.name === 'ie') {
+    // For IE we get a new window instead of a new tab, so we need to resize
+    // every time.
+    await driver.manage().window().setRect(config.windowSize);
```
Comment (rictic) [inline src/browser.ts]: oh subtle

File: src/browser.ts (hunk 1 above)
Comment (aomarks) [reply src/browser.ts]: :)

(actually it says >=0 since it's phrased the other way around)

**Pull request merged**